### DISCUSSION TITLE
How to compile d_example.cpp in the example directory?

### DISCUSSION THREAD

**Discussion by xiaorong117** [scivision/mumps-superbuild#90] [Q&A]
I was trying to compile d_example.cpp in the "example" directory.  I modified makefile to use mpicxx and mpif90  sequentially, but i failed.
	
 d_example: $$@.o
	mpif90 -o $@ $(OPTL) $@.o $(LIBDMUMPS) $(LORDERINGS) $(LIBS) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)

.SUFFIXES: .c .F .o .cpp
.F.o:
	$(FC) $(OPTF) -I. -I$(topdir)/include -I$(topdir)/src $(INCS) -c $*.F $(OUTF)$*.o
.c.o:
	$(CC) $(OPTC) $(CDEFS) -I. -I$(topdir)/include -I$(topdir)/src $(INCS) -c $*.c $(OUTC)$*.o
.cpp.o:
	mpicxx $(OPTC) $(CDEFS) -I. -I$(topdir)/include -I$(topdir)/src $(INCS) -c $*.cpp $(OUTC)$*.o

**Comment by scivision** [comment]
I'm not sure, I don't use the Makefiles, they are from the upstream MUMPS project https://mumps-solver.org
